### DISCUSSION TITLE
Add ability to record visitor information

### DISCUSSION THREAD

**Discussion by grantfitzsimmons** [specify/specify7#5717] [Feature Requests]
This could be done by extending/repurposing the Information Request table to allow capturing the visit information through a new relationship and table or by creating a new interactions table for this purpose.

See the attached PDF for more details

---

**Corinna P at CSIRO:**
> The now deprecated ANWC CMS had a module that allowed the capturing of information of visitors to the Collection. This is an important part of the business, and there are regular requests for statistics of this data for annual reports and the likes. We are currently trying to figure out how to continue this type of data capture.
> 
> Upon review of the underlying data structure (see attached PDF), we came to realise that a large portion of this can already be captured in Specify – the elements that are missing are a less restrictive infoRequest table (needs more ‘spare’ fields and linking to an object or recordset as optional), and a new table that allows capturing of information about the visit itself.
> 
> Is this something that you can see being integrated into Specify? We believe this would be a beneficial addition as, at least within CSIRO, all Collections should be capturing this information. I suspect other organisations could be interested in this too.
>
> [ANWC Visitor Info.pdf](https://github.com/specify/specify7/files/10457324/ANWC.Visitor.info-1.pdf)

@grantfitzsimmons:
> Does the current Information Request table fulfill some of your needs in Specify 7? It should not require a record set or collection object association when using 7 to create a new record and it has links to the agent table and dates. What restrictions are enforced in Specify 7 that would prevent you from using it for this purpose?

**Corinna P:**

> I wasn't aware that those restrictions (ie, need a linked CO or recordset) didn't apply in S7, so that's good to know. As per the diagram, yes, the InfoRequest and agent tables do translate roughly into Specify and we can most likely shoehorn the existing data into these tables. **However, the infoRequest table could do with a couple of spare fields to give the option to 'personalise' things a bit more.** 
>
> The crux really though is the ability to record visitor information, which will not be possible via the existing tables.

[Asana Link](https://app.asana.com/0/1201972579963460/1203699254683285/f)

**Comment by maxpatiiuk** [comment]
Could potentially be integrated with https://github.com/specify/specify7/issues/2601 to allow people to see the usage stats for their own deployment of specify 7

**Comment by emenslin** [comment]
Related to #1154

**Comment by g1rly-c0d3r** [comment]
Linked in https://github.com/specify/specify7/issues/8179
